velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -559,6 +561,76 @@ TpchPlan TpchQueryBuilder::getQ10Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ12Plan() const {
+  std::vector<std::string> ordersColumns = {"o_orderkey", "o_orderpriority"};
+  std::vector<std::string> lineitemColumns = {
+      "l_receiptdate",
+      "l_orderkey",
+      "l_commitdate",
+      "l_shipmode",
+      "l_shipdate"};
+
+  auto ordersSelectedRowType = getRowType(kOrders, ordersColumns);
+  const auto& ordersFileColumns = getFileColumnNames(kOrders);
+  auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+  const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId ordersScanNodeId;
+  core::PlanNodeId lineitemScanNodeId;
+
+  auto receiptDate = "l_receiptdate";
+  std::string receiptDateFilter = formatDateFilter(
+      receiptDate, lineitemSelectedRowType, "'1994-01-01'", "'1994-12-31'");
+  std::string commitDateFilter = "l_commitdate < l_receiptdate";
+  std::string shipDateFilter = "l_shipdate < l_commitdate";
```

Comment:
Did you check if these 2 filters can be applied as subfieldFilters in the TableScan ? In the plan below you added them as separate filter nodes. If they can be applied as subfieldFilters, is there a perf difference ?

Reply:
Yes, I initially tried having them as subfieldFilters, but when two or more columns are being compared in the filter, it cannot be in subfieldFilter (it could be in remainingFilter but only one of those can be added)
Parquet reader doesn't allow shipModeFilter to be in subfieldFilter and throws following error: 
Reason: Unsupported filter in parquet reader: Filter(MultiRange, deterministic, null not allowed)

Reply:
Thanks for confirming.